Source/Core/DiscIO/VolumeWiiCrypted.cpp

```diff
@@ -29,41 +29,105 @@
 
 namespace DiscIO
 {
-CVolumeWiiCrypted::CVolumeWiiCrypted(std::unique_ptr<IBlobReader> reader, u64 _VolumeOffset,
-                                     const unsigned char* _pVolumeKey)
-    : m_pReader(std::move(reader)), m_AES_ctx(std::make_unique<mbedtls_aes_context>()),
-      m_VolumeOffset(_VolumeOffset), m_last_decrypted_block(-1)
+constexpr u64 PARTITION_DATA_OFFSET = 0x20000;
+
+CVolumeWiiCrypted::CVolumeWiiCrypted(std::unique_ptr<IBlobReader> reader)
+    : m_pReader(std::move(reader)), m_game_partition(PARTITION_NONE), m_last_decrypted_block(-1)
 {
   _assert_(m_pReader);
 
-  mbedtls_aes_setkey_dec(m_AES_ctx.get(), _pVolumeKey, 128);
-}
+  // Get decryption keys for all partitions
+  CBlobBigEndianReader big_endian_reader(*m_pReader.get());
+  for (u32 partition_group = 0; partition_group < 4; ++partition_group)
+  {
+    u32 number_of_partitions;
+    if (!big_endian_reader.ReadSwapped(0x40000 + (partition_group * 8), &number_of_partitions))
+      continue;
 
-bool CVolumeWiiCrypted::ChangePartition(u64 offset)
-{
-  m_VolumeOffset = offset;
+    u32 read_buffer;
+    big_endian_reader.ReadSwapped(0x40000 + (partition_group * 8) + 4, &read_buffer);
+    u64 partition_table_offset = (u64)read_buffer << 2;
 
-  u8 volume_key[16];
-  DiscIO::VolumeKeyForPartition(*m_pReader, offset, volume_key);
-  mbedtls_aes_setkey_dec(m_AES_ctx.get(), volume_key, 128);
-  return true;
+    for (u32 i = 0; i < number_of_partitions; i++)
+    {
+      big_endian_reader.ReadSwapped(partition_table_offset + (i * 8), &read_buffer);
+      u64 partition_offset = (u64)read_buffer << 2;
+
+      if (m_game_partition == PARTITION_NONE)
+      {
+        u32 partition_type;
+        if (big_endian_reader.ReadSwapped(partition_table_offset + (i * 8) + 4, &partition_type))
```

Comment:
Can you put some braces here? At least on this outer `if`, because theres 2 levels inside which could be hard to read when extended in the future.